src/modules/0.0.21/aws_dynamo/index.ts

```diff
@@ -146,7 +156,10 @@ class DynamoTableMapper extends MapperBase<DynamoTable> {
           } as WaiterOptions<DynamoDB>,
           { TableName: e.tableName },
         );
-        const newTable = this.dynamoMapper(res.TableDescription);
+        const newTable = this.dynamoMapper(
+          res.TableDescription,
+          (e.region as AwsRegions)?.region ?? (e.region as string),
```

Comment:
why `e.region` is not enough here? is it not a string?

Reply:
So, I was unable yesterday to get TypeORM to generate a foreign key relation to the `aws_regions` table and also have a default value without some weird trickery with the Typescript type.

The *behavior* of TypeORM, even if you specified only `AwsRegions` as the type, is to return a `string`, but if you set the type to `string` then it fails to generate the foreign key relation for some reason, so I did a weird `string | AwsRegions` type and now the code above has to probe for the `region` string to make Typescript happy.

I do want to try to simplify this again, since this "pattern" could end up in a lot of modules and it is really clunky, but I am not sure I can make it any better than it currently is without manually editing the migration files, which is also suboptimal.

Reply:
it is not optimal but could be a trigger after insert/update, if no region, set default. let me see if I can think about something else

Reply:
No, this really does work and it has Postgres set the default value exactly like we want.

We're in the codebase all the time and have the mental model in our heads, the users will presumably interact with IaSQL, even in the best of cases, a far smaller amount of time, so the more weirdness exposed to them the harder it will be to adopt.

I am digging through the TypeORM decorator's actual implementation to see if I can find another way.

Reply:
did a quick search and seems to be an open issue not expected to be solved https://github.com/typeorm/typeorm/issues/2427